packages/core/src/setup.ts

```diff
@@ -61,7 +61,7 @@ type ToProvidedActor<
     id: IsNever<TChildrenMap> extends true
       ? string | undefined
       : K extends keyof Invert<TChildrenMap>
-        ? Invert<TChildrenMap>[K]
+        ? Invert<TChildrenMap>[K] & string
```

Comment:
this is the true and correct fix for what I previously fixed by removing `Compute`. You can read my breakdown of the issue [here](https://github.com/microsoft/TypeScript/issues/57188#issuecomment-1926882621)